### PR TITLE
Get estimated memory consumption of an element

### PR DESCRIPTION
**Pull request by GuyAv46**
added a function to return expected element memory consumption according to the given parameters, including metadata for it.

### CODE REVIEW

File: src/VecSim/vec_sim.h [hunk 1]
```diff
@@ -27,6 +27,14 @@ VecSimIndex *VecSimIndex_New(const VecSimParams *params);
  */
 size_t VecSimIndex_EstimateInitialSize(const VecSimParams *params);
 
+/**
+ * @brief normalize the vector blob in place.
```
Comment (alonre24) [inline src/VecSim/vec_sim.h:31]: Remove...

**Review by alonre24 — COMMENTED** [review]
LGTM, one small fix

File: src/VecSim/algorithms/brute_force/brute_force.cpp [hunk 2]
```diff
@@ -49,6 +49,10 @@ size_t BruteForceIndex::estimateInitialSize(const BFParams *params) {
     return est;
 }
 
+size_t BruteForceIndex::estimateElementMemory(const BFParams *params) {
+    return params->dim * sizeof(float);
```
Comment (alonre24) [inline src/VecSim/algorithms/brute_force/brute_force.cpp:53]: Actually, why not consider the meta-data as well? size of `VectorBlockMember`, and `labelToIdLookup`entry?

File: tests/unit/test_hnswlib.cpp [hunk 3]
```diff
@@ -1379,10 +1379,13 @@ TEST_F(HNSWLibTest, testSizeEstimation) {
     size_t size_data_per_element = size_links_level0 + dim * sizeof(float) + sizeof(labeltype);
     test_estimation += n * size_data_per_element;
 
-    size_t estimation = VecSimIndex_EstimateInitialSize(&params);
-    ASSERT_GE(estimation, test_estimation);
+    size_t iestimation = VecSimIndex_EstimateInitialSize(&params);
+    ASSERT_GE(iestimation, test_estimation);
```
Comment (alonre24) [inline tests/unit/test_hnswlib.cpp:1383]: In second thought, maybe a better test will be comparing the estimation to the actual memory consumption as reported by the allocator (relevant for BF as well).

**Review by alonre24 — COMMENTED** [review]
I suggest comparing the estimated memory consumption (both for the index initial memory and the element) to the actual memory report by the allocator.

**Review by DvirDukhan — APPROVED** [review]

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**